data/src/test/java/org/apache/iceberg/data/DataTestHelpers.java

```diff
@@ -44,7 +43,7 @@ public static void assertEquals(Types.StructType struct, Record expected, Record
   public static void assertEquals(Types.ListType list, List<?> expected, List<?> actual) {
     Type elementType = list.elementType();
 
-    Assert.assertEquals("List size should match", expected.size(), actual.size());
+    assertThat(actual).as("List size should match").hasSize(expected.size());
```

Comment:
```suggestion
    assertThat(actual).as("List size should match").hasSameSizeAs(expected);
```
when this assertion fails it will also print the content so that it's easier to understand why the assertion fails

Reply:
cool, thx, done